### ISSUE TITLE
Infinity values for closeness centrality in connected graph

### ISSUE THREAD

**Issue by johnymontana** [neo4j-contrib/neo4j-graph-algorithms#291]
In a connected graph, my understanding is that closeness centrality should always be calculatable, and only in a disconnected graph is it possible to have closeness centralities evaluated at NaN/infinity. 

However, when running closeness centrality on a small connected graph, several closeness centralities result in "infinity":

Steps to reproduce:

~~~
LOAD CSV WITH HEADERS FROM "https://raw.githubusercontent.com/mathbeveridge/asoiaf/master/data/asoiaf-all-edges.csv" AS row
MERGE (src:Character {name: row.Source})
MERGE (tgt:Character {name: row.Target})
// global relationship
MERGE (src)-[r:INTERACTS]->(tgt) ON CREATE SET r.weight = toInteger(row.weight)
~~~

~~~
CALL algo.closeness('Character', 'INTERACTS', {write:true, writeProperty:"closeness", stats:true})
~~~

Query for closeness, order by DESC:

~~~
neo4j> MATCH (c:Character) WHERE exists(c.closeness) AND c.closeness IS
 NOT null
       RETURN c.name AS name, c.closeness AS closeness
       ORDER BY closeness DESC LIMIT 25
       ;
+------------------------------------------------+
| name                               | closeness |
+------------------------------------------------+
| "Addam-Marbrand"                   | Infinity  |
| "Aegon-Frey-(son-of-Stevron)"      | Infinity  |
| "Aegon-I-Targaryen"                | Infinity  |
| "Aegon-Targaryen-(son-of-Rhaegar)" | Infinity  |
| "Aegon-V-Targaryen"                | Infinity  |
| "Aemon-Targaryen-(Dragonknight)"   | Infinity  |
| "Aenys-Frey"                       | Infinity  |
| "Aeron-Greyjoy"                    | Infinity  |
| "Aerys-I-Targaryen"                | Infinity  |
| "Aerys-II-Targaryen"               | Infinity  |
| "Aggar"                            | Infinity  |
| "Aggo"                             | Infinity  |
| "Alayaya"                          | Infinity  |
| "Amabel"                           | Infinity  |
~~~

**Comment by mknblch** [comment]
Thanks for your feedback! I tested your Graph and it looks like there are some disconnected nodes. Closeness on a disconnected node evaluates to Infinity due to `N-1 / (double) 0`. As far as i know this should by convention evaluate to 0. The bug is addressed in #310

**Comment by mknblch** [comment]
see #310
